iree_tests/README.md

```diff
@@ -0,0 +1,111 @@
+# IREE Tests
+
+This directory contains generated test suites for running through IREE's
+compiler and runtime tools.
+
+Each test suite has one folder per test program containing a few files:
+
+```
+[program name 1]/
+  model.mlir
+  input_0.npy
+  output_0.npy
+  test_data_flags.txt
+```
+
+Where:
+
+* `model.mlir` is in a format that is ready for use with `iree-compile`
+  (e.g. torch-mlir, stablehlo, tosa, linalg)
+* `input_0.npy` and `output_0.npy` files correspond to any number of program
+  inputs and outputs for one test case
+* `test_data_flags.txt` is a flagfile for use with
+  `iree-run-module --flagfile=test_data_flags.txt` of the format:
+
+  ```
+  --input=@input_0.npy
+  --expected_output=@output_0.npy
+  ```
+
+## Running the test suite
+
+Developer quickstart:
+
+```bash
+python -m pip install iree-compiler iree-runtime
+python -m pip install -r ./iree_tests/onnx/requirements.txt
+
+python ./iree_tests/onnx/import_tests.py
+python ./iree_tests/compile_tests.py
+python ./iree_tests/run_tests.py
```

Comment:
At least the compile and run scripts will likely be replaced with pytest files / plugins.

* https://docs.pytest.org/en/stable/
* https://docs.pytest.org/en/stable/how-to/writing_plugins.html#conftest-py-plugins
* https://docs.pytest.org/en/stable/example/nonpython.html#yaml-plugin

Reply:
Here's a GitHub Actions workflow to run the 'compile' and 'run' scripts: https://github.com/ScottTodd/SHARK-TestSuite/blob/test-githubaction/.github/workflows/test_iree.yml

Sample runs:
* https://github.com/ScottTodd/SHARK-TestSuite/actions/runs/8024507737/job/21923232663 (all tests, 'run' is hanging for some reason on the Linux server...)
* https://github.com/ScottTodd/SHARK-TestSuite/actions/runs/8024846159/job/21924226218 (limited to 20 tests)

I'm going to set up pytest next so we can get better test output, define run configurations, check XFAIL/XPASS, etc.

Reply:
and conftest.py that replaces the `compile_tests.py` script (`run_tests.py` port coming soon too): https://github.com/ScottTodd/SHARK-TestSuite/blob/iree-pytest/iree_tests/onnx/basic/conftest.py

posted a more detailed update on how that fits together here: https://github.com/openxla/iree/issues/16372#issuecomment-1962143140